compress/zstd.go

```diff
@@ -0,0 +1,21 @@
+package compress
+
+import "github.com/klauspost/compress/zstd"
+
+type compressorZstd struct {
+	encoder *zstd.Encoder
+	decoder *zstd.Decoder
+}
+
+func (c *compressorZstd) Compress(src []byte) ([]byte, error) {
+	return c.encoder.EncodeAll(src, nil), nil
+}
+
+func (c *compressorZstd) Decompress(src []byte) ([]byte, error) {
+	return c.decoder.DecodeAll(src, nil)
+}
+
+func (c *compressorZstd) Close() error {
+	c.decoder.Close()
+	return c.encoder.Close()
+}
```

Comment:
Should we make this unexported since a higher interface level (`Compressor` struct) is present? Same for `compressorZstdCgo` as well?
```suggestion
func (c *compressorZstd) compress(src []byte) ([]byte, error) {
	return c.encoder.EncodeAll(src, nil), nil
}

func (c *compressorZstd) decompress(src []byte) ([]byte, error) {
	return c.decoder.DecodeAll(src, nil)
}

func (c *compressorZstd) close() error {
	c.decoder.Close()
	return c.encoder.Close()
}
```

Reply:
You can't use them anyway because the struct is not exported, right? But if you keep them exported it gives you the ambiguity that forces you to define a parent method because both children have the same method. Anyway, either way it works and the end user can't use them so :man_shrugging: